src/v/cluster/topic_table_probe.cc

```diff
@@ -16,12 +16,13 @@
 #include "config/node_config.h"
 #include "metrics/metrics.h"
 #include "metrics/prometheus_sanitize.h"
+#include "model/fundamental.h"
 
 namespace cluster {
 
 topic_table_probe::topic_table_probe(const topic_table& topic_table)
   : _topic_table(topic_table)
-  , _node_id(*config::node().node_id()) {
+  , _node_id(config::node().node_id().value_or(model::unassigned_node_id)) {
```

Comment:
wondering if checked access using value() is the right way because node_id is assumed to be set by this point and using an unassigned_node_id has no meaning here.

Reply:
Yeah I can do that but will need to fix a bunch of tests